third_party/protobuf/BUILD.gn

```diff
@@ -23,6 +23,8 @@ config("protobuf_config") {
         "-Wno-enum-compare",
         # Older version of clang don't have -Wno-enum-compare-switch
         "-Wno-unknown-warning-option",
+
+        "-Wno-deprecated-declarations",
```

Comment:
What is this needed for? Should there be a cleanup bug for this?

Reply:
I ran into a compiler error when i switched to cpp20 -  but maybe that was because of a misconfiguration. ill try removing this , rerunning and seeing what happens.

Reply:
removed, tried a small test , saw it works locally. Added a new commit to remove it